data-analytics/dataflow-python-examples/dataflow_python_examples/image_labels.py

```diff
@@ -0,0 +1,148 @@
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+image_labels.py is a Dataflow pipeline which reads a file
+containing web urls or gcs references to image files and writes
+the original file reference to a BigQuery table with the labels
+identified by the Cloud Vision API.
+"""
+
+from __future__ import absolute_import
+import argparse
+import logging
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions
+from google.cloud import vision
+from google.cloud.vision import types
+
+# Specify default parameters.
+INPUT_FILE = 'gs://python-dataflow-example/data_files/image-list.txt'
+BQ_DATASET = 'ImageLabelFlow'
+BQ_TABLE = BQ_DATASET + '.dogs_short'
+
+
+def detect_labels_uri(uri):
+    """This Function detects labels in the image file located in Google Cloud Storage or on
+    theWeb and returns a comma separated list of labels. This will return an empty string
+    if not passed a valid image file
+
+    Args:
+        uri: a string link to a photo in gcs or on the web
+
+    (Adapted From:
+    https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/vision/cloud-client/detect/detect.py)
+    """
+    # Initialize cloud vision api client and image object.
+    client = vision.ImageAnnotatorClient()
+    image = types.Image()
+    image.source.image_uri = uri
+
+    # Send an api call for this image, extract label descriptions
+    # and return a comma-space separated string.
+    response = client.label_detection(image=image)
+    labels = response.label_annotations
+    label_list = [l.description for l in labels]
+    return ', '.join(label_list)
+
+
+class ImageLabeler(beam.DoFn):
+    """
+        This DoFn wraps the inner function detect_labels_uri to write a BigQuery row dictionary
+        with the image file reference specified by the element from the prior PCollection and a list
+        of labels for that image assigned by the Google Cloud Vision API.
+    """
+
+    def process(self, element, *args, **kwargs):
+        """
+        Args:
+            element: A string specifying the uri of an image
+
+        Returns:
+            row: A list containing a dictionary defining a record to be written to BigQuery
+        """
+        labels_string = detect_labels_uri(element)
+        row = [{'image_location': element, 'labels': labels_string}]
```

Comment:
Inlining is good, instead of creating a new variable

`return [{'image_location': element, 'labels': detect_labels_uri(element)}]`

Reply:
Removing unnecessary assignment.